controllers/semeru_compiler.go

```diff
@@ -0,0 +1,148 @@
+
+	deploy.Labels = getLabels(wlva)
+	deploy.Spec.Strategy.Type = appsv1.RecreateDeploymentStrategyType
+	replicas := int32(1)
+	deploy.Spec.Replicas = &replicas
+
+	if deploy.Spec.Selector == nil {
+		deploy.Spec.Selector = &metav1.LabelSelector{
+			MatchLabels: getSelectors(wlva),
+		}
+	}
+
+	deploy.Spec.Template = corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   wlva.Name + SemeruLabelName,
+			Labels: getLabels(wlva),
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:            JitServer,
+					Image:           wlva.Status.GetImageReference(),
+					ImagePullPolicy: *wlva.GetPullPolicy(),
+					Command:         []string{"jitserver"},
+					Ports: []corev1.ContainerPort{
+						{
+							ContainerPort: 38400,
+							Protocol:      corev1.ProtocolTCP,
+						},
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceMemory: resource.MustParse("1200Mi"),
+							corev1.ResourceCPU:    resource.MustParse("1000m"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceMemory: resource.MustParse("1200Mi"),
+							corev1.ResourceCPU:    resource.MustParse("8000m"),
+						},
+					},
+					Env: []corev1.EnvVar{
+						{Name: "OPENJ9_JAVA_OPTIONS", Value: "-XX:+JITServerLogConnections"},
+					},
+				},
+			},
+		},
+	}
+}
+
+func reconcileSemeruService(svc *corev1.Service, wlva *wlv1.WebSphereLibertyApplication) {
+	var port int32 = 38400
+	var timeout int32 = 86400
+	svc.Labels = getLabels(wlva)
+	svc.Spec.Selector = getSelectors(wlva)
+	if len(svc.Spec.Ports) == 0 {
+		svc.Spec.Ports = append(svc.Spec.Ports, corev1.ServicePort{})
+	}
+	svc.Spec.Ports[0].Protocol = corev1.ProtocolTCP
+	svc.Spec.Ports[0].Port = port
+	svc.Spec.Ports[0].TargetPort = intstr.FromInt(int(port))
+	svc.Spec.SessionAffinity = corev1.ServiceAffinityClientIP
+	svc.Spec.SessionAffinityConfig = &corev1.SessionAffinityConfig{
+		ClientIP: &corev1.ClientIPConfig{
+			TimeoutSeconds: &timeout,
+		},
+	}
+}
+
+// Create the Selector map for a Semeru Compiler
+func getSelectors(wlva *wlv1.WebSphereLibertyApplication) map[string]string {
+	requiredSelector := make(map[string]string)
+	requiredSelector["app.kubernetes.io/component"] = SemeruLabelName
+	requiredSelector["app.kubernetes.io/instance"] = wlva.GetName() + SemeruLabelNameSuffix
+	requiredSelector["app.kubernetes.io/part-of"] = wlva.GetName()
+	return requiredSelector
+}
+
+// Create the Labels map for a Semeru Compiler
+func getLabels(wlva *wlv1.WebSphereLibertyApplication) map[string]string {
+	requiredLabels := make(map[string]string)
+	requiredLabels["app.kubernetes.io/name"] = wlva.GetName() + SemeruLabelNameSuffix
+	requiredLabels["app.kubernetes.io/instance"] = wlva.GetName() + SemeruLabelNameSuffix
+	requiredLabels["app.kubernetes.io/managed-by"] = `websphere-liberty-operator`
```

Comment:
Also: `requiredLabels["app.kubernetes.io/managed-by"] = OperatorName`

Reply:
Yes, it'll be good to use constant that already exists